c_src/filewatch_inotify.c

```diff
@@ -24,21 +26,16 @@ enum {
 
 /*
   A struct inotify_event's relevant data is copied to an array of
-  struct message. This fixes a bug where messages sent back to Erlang
+  struct Message. This fixes a bug where messages sent back to Erlang
   had incorrect filenames.
 */
-struct message {
-    int wd;
-    size_t len;
-    char name[NAME_MAX];
-};
-
 struct instance {
     ErlDrvPort port;
     int fd;
     size_t n_events;
     unsigned long cooldown_ms;
-    struct message msgs[MAX_EVENTS];
+    struct Message msgs[MAX_EVENTS];
```

Comment:
I find it a bit strange that you typedef-ed the `Message` struct yet you still use the `struct` keyword when using the struct. The original reason for typedef-ing structs was that crappy old compiler (*cough* visual studio *cough*) couldn't handle the struct keyword in variable definitions (later cargo-culted by lazy programmers that wanted to save on 6 letters).

In other words, if you continue to use the `struct` keyword then there's no reason to typedef the struct. Note that if you want to forward declare a struct, you can simply write `struct message;` and that's it.